File: sem3/map/Lab/2/src/model/AssignStmt.java
package model;

public class AssignStmt implements IStmt {
    private String id;
    private Exp exp;

    public AssignStmt(String id, Exp exp) {
        this.id = id;
        this.exp = exp;
    }

    @Override
    public String toString() {
        return this.id + " = " + exp.toString();
    }

    @Override
    public PrgState execute(PrgState state) {
        MyIStack <IStmt> stk = state.getStk();
        MyIDictionary<String, Integer> symTable = state.getSymTable();
        symTable.put(this.id, this.exp.eval(symTable, state.getHeap()));
        return null;
    }
}

File: sem3/map/~craciunf/c-sharp-toy-language-interpreter/models/AssignStmt.cs
/* package models; */
using System;
using System.Collections;
using System.Collections.Generic;

namespace ToyLanguageInterpreter {
    public class AssignStmt : IStmt {
        private String id;
        private Exp exp;

        public AssignStmt(String id, Exp exp) {
            this.id = id;
            this.exp = exp;
        }

        /* @Override */
        public String toString() {
            return id + " = " + exp.toString();
        }

        /* @Override */
        public PrgState execute(PrgState state) {
            // MyIStack <IStmt> stack = state.getExeStack();
            Dictionary <String, int> symTable = state.getSymTable();

            int val = this.exp.eval(symTable);
            // if(symTable.containsKey(id))
            //     symTable.put(id, val);
            // else
            symTable.Add(this.id, val);

            return state;
        }
    }
}
